src/Tickets/QR/Settings.php

```diff
@@ -0,0 +1,165 @@
+
+	/**
+	 * Get the option key that will be used to store the value.
+	 *
+	 * Option keys that are user controlled, need to be kept as public.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public static function get_enabled_key(): string {
+		return static::$enabled_key;
+	}
+
+	/**
+	 * Get the option key that will be used to store the value.
+	 *
+	 * Option keys that are not user controlled, should be kept as protected and interactions should be using
+	 * the methods related, like `$this->get_api_hash()`, avoids weird non-tested scenarios.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	protected static function get_api_hash_key(): string {
+		return static::$api_hash_key;
+	}
+
+	/**
+	 * Check if the QR code is enabled.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed $context The context of the check.
+	 *
+	 * @return bool
+	 */
+	public function is_enabled( $context = null ): bool {
+		$controller = tribe( Controller::class );
+		$enabled = false;
+
+		// Only fetch from DB if the controller can use the QR code.
+		if ( $controller->can_use() && $controller->has_library_loaded() ) {
+			$enabled = tribe_is_truthy( tribe_get_option( static::get_enabled_key(), false ) );
+		}
+
+		/**
+		 * Filters the QR enabled value
+		 *
+		 * @since 5.6.10
+		 * @deprecated TBD Use tec_tickets_qr_code_enabled instead.
+		 *
+		 * @param bool  $enabled The bool that comes from the options
+		 * @param array $context Context for this check, normally an Array with the ticket
+		 */
+		$enabled = apply_filters_deprecated( 'tribe_tickets_plus_qr_enabled', [ $enabled, $context ], 'TBD', '' );
+
+		/**
+		 * Filters the QR enabled value.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool  $enabled The bool that comes from the options.
+		 * @param array $context Context for this check, normally an Array with the ticket.
+		 */
+		return apply_filters( 'tec_tickets_qr_code_enabled', $enabled, $context );
+	}
+
+	/**
+	 * Generate a hash key for QR API.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The QR API hash.
+	 */
+	protected function generate_api_hash(): string {
+		$random = wp_generate_password( 24, true, true );
+		$hash   = substr( md5( $random ), 0, 8 );
+
+		/**
+		 * Filters the generated hash key for QR API.
+		 *
+		 * @since      4.7.5
+		 *
+		 * @deprecated TBD Use tec_tickets_qr_settings_generated_api_hash instead.
+		 *
+		 * @param string $api_key a API key string.
+		 */
+		$hash = apply_filters_deprecated( 'tribe_tickets_plus_qr_api_hash', [ $hash ], 'TBD', 'tec_tickets_qr_settings_generated_api_hash' );
+
+		/**
+		 * Filters the generated hash key for QR API.
+		 *
+		 * @since TBD
+		 *
+		 * @param string $hash The API hash string.
+		 */
+		return apply_filters( 'tec_tickets_qr_settings_generated_api_hash', $hash );
+	}
+
+	/**
+	 * Get the API hash key, if none exists, generate one.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_api_hash(): string {
```

Comment:
I find this confusing, it's the api key you're getting by using the method. And the term we use for it is Api Key. Naming this api hash makes it more confusing than anything else.

Reply:
So in the previous code there was two different terminologies, I picked hash because that is what we do with the method, we create a md5 hash of a random password. Now if we want to go with `get_api_key` we can go that route too.